### PR TITLE
Improved Versioning and Error Handling

### PR DESCRIPTION
**Pull request by mplanchard**
This commit contains some useful additions, including a `version.py`
for tracking the version number and automatically pulling into
`setup.py` and the `__init__` file. Also improved the error message
if None is passed into `compare_packages`. Added tox support for
python 2.6, 2.7, 3.4, and 3.5.
- pep compliance
- version number
- added rpm module to `__init__` import
- Added TypeError check when parsing versions to give useful
  output if someone accidentally passes a None
- added `setup.cfg` with universal flag
- fixed `setup.cfg`
- Added tox support
- Updated imports in test_common
- Added `version.py`
- Import `__version__` and `__version_info__` at package level
- `setup.py` uses `version.py` to get version
- Updated package level documentation in `__init__.py`
- Moved stuff around a bit in `__init__.py`

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -65,9 +65,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = '0.1.0'
+version = '0.2.0'
 # The full version, including alpha/beta/rc tags.
-release = '0.1.0'
+release = '0.2.0'
```
Comment (erinbleiweiss) [inline docs/conf.py]: These version numbers don't match the number in version.py (0.2.0 ≠ 0.2.2)

**Comment by erinbleiweiss** [comment]
Looks good to me, aside from a possible version number issue

File: docs/conf.py (hunk 1 above)
Comment (mplanchard) [inline docs/conf.py]: Oh yeah. I always forget about Sphinx versioning. I'll try to get that to pull automatically, too.

**Comment by mplanchard** [comment]
Check it out. Seems to work (you can check locally by running `make html` from the `docs` directory)

**Comment by erinbleiweiss** [comment]
Cool, I approve!

**Pull request merged**